generators/base-generator.js

```diff
@@ -117,7 +117,9 @@ module.exports = yeoman.Base.extend({
             return prompt.commonFilter.call(this, input);
           }.bind(this);
         }
-        if (!prompt.hasOwnProperty('validate') && prompt.hasOwnProperty('name')) {
+        if (prompt.hasOwnProperty('validate')) {
+          newPrompt.validate = newPrompt.validate.bind(this);
```

Comment:
I modified this code to only bind 'this' with validate method. I am not sure if this is the correct way to handle this.